### PR TITLE
Handle dist-git-sources for gitlab

### PR DESCRIPTION
**Pull request by lukaszachy**
Fixes: #1613

### CODE REVIEW

File: tmt/utils.py [hunk 1]
```diff
@@ -2780,6 +2780,15 @@ class CentOSDistGit(DistGitHandler):
     remote_substring = re.compile(r'redhat/centos')
 
 
+class RHGitlab(DistGitHandler):
```
Comment (psss) [inline tmt/utils.py:2783]:
```suggestion
class RedHatGitlab(DistGitHandler):
```

It's just four extra letters :)

File: tests/unit/test_utils.py [hunk 2]
```diff
@@ -562,31 +562,40 @@ def test_run_big():
...
-
-def test_FedoraDistGit(tmpdir):
-    # Fake values, production hash is too long
-    tmpdir.join('sources').write(
-        'SHA512 (fn-1.tar.gz) = 09af\n')
-    tmpdir.join('tmt.spec').write('')
-    fedora_sources_obj = tmt.utils.FedoraDistGit()
-    assert [("https://src.fedoraproject.org/repo/pkgs/rpms/tmt/fn-1.tar.gz/sha512/09af/fn-1.tar.gz",  # noqa: E501
-            "fn-1.tar.gz")] == fedora_sources_obj.url_and_name(cwd=tmpdir)
+class Test_dist_git_source:
```
Comment (psss) [inline tests/unit/test_utils.py:565]: Why are you adding the class here? I remember @happz just recently started removing classes from unit tests as they are not needed.

File: tmt/utils.py [hunk 3]
```diff
@@ -2780,6 +2780,15 @@ class CentOSDistGit(DistGitHandler):
...
+    """ Red Hat on Gitlab """
+    usage_name = "rhgitlab"
```
Comment (psss) [inline tmt/utils.py:2785]:
```suggestion
    usage_name = "redhatgitlab"
```

Similar as above.

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for fixing this. Looks good and works nicely! Just a few minor suggestions/questions.

**Comment by psss** [comment]
Package buidling fails with the following:

```
tmt.utils.GeneralError: No known remote in 'remote.origin.url https://<redacted_credentials>@gitlab.com/redhat/rhel/rpms/osbuild.git'.
```

File: tests/unit/test_utils.py (hunk 2 above)
Comment (lukaszachy) [reply tests/unit/test_utils.py:565]: Easier to navigate and filter test cases. But frankly I don't care so I'll undo

**Comment by lukaszachy** [comment]
> Package buidling fails with the following:
> 
> ```
> tmt.utils.GeneralError: No known remote in 'remote.origin.url https://<redacted_credentials>@gitlab.com/redhat/rhel/rpms/osbuild.git'.
> ```

Eh, list vs str :/

**Review by psss — APPROVED** [review]

**Pull request merged**